Luna/Luna/SettingsPeriodsViewController.swift

```diff
@@ -17,16 +17,34 @@ class SettingsPeriodsViewController: UIViewController, UITableViewDelegate, UITa
         
         tableView.delegate = self
         tableView.dataSource = self
-        
-        periodArray = periodArrayRecieved
-        
-        // Do any additional setup after loading the view.
+    }
+    
+    override func viewWillAppear(_ animated: Bool)
+    {
+        //Need to reload the period objects after they have been editted
+        loadPeriodsArray()
     }
 
     override func didReceiveMemoryWarning()
     {
         super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    }
+    
+    
+    func loadPeriodsArray()
+    {
+        settingsPeriodViewModel.getPeriods()
+        {
+            errorOrNil in
+            
+            guard errorOrNil == nil else
+            {
+                return
+            }
+            
+            //Reload the table after retrieving the period objects from Firebase
+            self.tableView.reloadData()
```

Comment:
Uh oh. This is definitely a blocker! 😬 

This needs to be called as such: `DispatchQueue.main.async { self.tableView.reloadData() }`

We cannot update a UI element from a background thread. We just jump back to the main thread, which is where the UI ops run.